src/Behat/Cores/Traits/FileTrait.php

```diff
@@ -32,7 +37,7 @@ public function createFileWithName($file_path, $directory = NULL) {
       $data = file_get_contents($file_path);
 
       // Existing files are replaced.
-      $file = file_save_data($data, $destination, 1);
+      $file = $this->getFileRepository()->writeData($data, $destination, 1);
```

Comment:
This library is still compatible with Drupal 7 and it must remain like that until we decide to don't support Drupal 7 anymore. 

This change breaks the Drupal 7 compatibility as the file repository service does not exist.

Here is my suggestion:

```suggestion
      $file = function_exists('file_save_data') ? file_save_data($data, $destination, 1)  : $this->getFileRepository()->writeData($data, $destination, 1);
```

Reply:
Wait not, ignore that suggestion, I think this must be done in other way:

1. Create an abstract method to save data (s.e.: `fileSaveData`). It must receive the same three arguments.
2. Implement the method in Drupal8 class by doing this changes you have just done. The property and the method to get the file repository must be moved to that class.
3. Implement the method in Drupal7 class by calling file_save_data.